agent/include/content_analysis/sdk/analysis_agent.h

```diff
@@ -115,22 +165,21 @@ class Agent {
   };
 
   // Returns a new agent instance and calls Start().
-  static std::unique_ptr<Agent> Create(Config config);
+  static std::unique_ptr<Agent> Create(
+      Config config, std::unique_ptr<AgentEventHandler> handler);
 
   virtual ~Agent() = default;
 
-  // Returns the configuration parameters of the agent.
+  // Returns the configuration parameters used to create the agent.
   virtual const Config& GetConfig() const = 0;
 
-  // Returns a new content analysis session from Google Chrome.
-  // The session's associated request and response have already been
-  // properly initialized and are ready for use.
-  virtual std::unique_ptr<Session> GetNextSession() = 0;
+  // Handles events triggered on this agent and calls the coresponding
+  // callbacks in the AgentEventHandler.  This method is blocking and returns
+  // when Stop() is called or if an error occurs.
+  virtual void HandleEvents() = 0;
 
-  // Prepares the agent for graceful shutdown.  A function blocked on
-  // GetNextSession() will return with a null session.  While Agent is
-  // generally not thread safe, this method is the exception and may be
-  // called from any thread.
+  // Prepares the agent for graceful shutdown.  Any function blocked on
+  // HandleEvents() will return.
   virtual int Stop() = 0;
```

Comment:
Good idea.  I updated the comments to say 0 is success and -1 is failure since that is what the code does.  It would be nice to start defining specific error codes, and when we do that we can update the comments to reflect the action error codes returned.  Sound good?

Reply:
Yup, sounds good!